medusa/config.py

```diff
@@ -216,6 +216,7 @@ def parse_config(args, config_file):
     config.set('cassandra', 'resolve_ip_addresses', 'True'
                if evaluate_boolean(config['cassandra']['resolve_ip_addresses']) else 'False')
     kubernetes_enabled = evaluate_boolean(config['kubernetes']['enabled'])
+    grpc_enabled = evaluate_boolean(config['grpc']['enabled'])
```

Comment:
I'm confused by the link to server_test.py because it actually links to the integration_steps.py.

I had to touch the grpc enablement config because of the bug I explained in the comment below.

The issue is that we have 3 or 4 code paths of how the MedusaConfig gets built. 
- There's the vanila one for when Medusa runs [here](https://github.com/thelastpickle/cassandra-medusa/blob/master/medusa/config.py#L294). The args are stuff that comes directly from CLI options.
- When Medusa starts with the gRPC server, it actually creates a config in the [server's constructor](https://github.com/thelastpickle/cassandra-medusa/blob/master/medusa/service/grpc/server.py#L53), but also in the subsequent [serve() method](https://github.com/thelastpickle/cassandra-medusa/blob/master/medusa/service/grpc/server.py#L67). Bot of these call the `create_config()` which then calls the same load_config() as the vanilla scenario, except they pass no args as there is no CLI.

This looks like the same code path, so so far so good.

In the integration tests, the base case is this [get_args()](https://github.com/thelastpickle/cassandra-medusa/blob/master/tests/integration/features/steps/integration_steps.py#L437). This [ends up calling the same load_config()](https://github.com/thelastpickle/cassandra-medusa/blob/master/tests/integration/features/steps/integration_steps.py#L499) as the vanilla case, except the args we pass here are not key-values comming from the CLI, but a merged dict that looks more like the actual config proper. Looking at it now, this does not feel too good.

When we run ITs with grpc, we seem to [parse](https://github.com/thelastpickle/cassandra-medusa/blob/master/medusa/config.py#L176) config instead of loading it. The difference seems to be the _parse_ does the actual reading, and the _load_ does first call parse, then throws if something is not in order.

At any rate, the ITs with grpc server do construct MedusaConfig again after getting it, [maybe to ommit some secions](https://github.com/thelastpickle/cassandra-medusa/blob/master/tests/integration/features/steps/integration_steps.py#L373).

ITs with grpc server and mgmt api seem to do the same thing as the ones with just grpc server.

So to somehow sum this up. I think we're mis-using the `args` in the tests. We are putting things there that ought not to be there (you cannot control grpc on/off via CLI) but that's precisely what we do. I would not touch this in this PR, but perhaps make a refactoring ticket to clean this up. WDYT?

But, the issue with `enabled` -> `k8s|grpc_enabled` is a real bug and I had to do it.

Reply:
Thanks for the explanation.